internal/file/fingerprint.go

```diff
@@ -0,0 +1,235 @@
+}
+
+const (
+	OutputFileNameFingerprints = ".debricked.fingerprints.wfp"
+)
+
+func isExcludedFile(filename string) bool {
+
+	return isExcludedByExtension(filename) ||
+		isExcludedByFilename(filename) ||
+		isExcludedByEnding(filename)
+}
+
+func isExcludedByExtension(filename string) bool {
+	filenameLower := strings.ToLower(filename)
+	for _, format := range EXCLUDED_EXT {
+		if filepath.Ext(filenameLower) == format {
+			return true
+		}
+	}
+
+	return false
+}
+
+func isExcludedByFilename(filename string) bool {
+	filenameLower := strings.ToLower(filename)
+	for _, file := range ECLUDED_FILES {
+		if filenameLower == file {
+			return true
+		}
+	}
+
+	return false
+}
+
+func isExcludedByEnding(filename string) bool {
+	filenameLower := strings.ToLower(filename)
+	for _, ending := range EXCLUDED_FILE_ENDINGS {
+		if strings.HasSuffix(filenameLower, ending) {
+			return true
+		}
+	}
+
+	return false
+}
+
+type IFingerprint interface {
+	FingerprintFiles(rootPath string, exclusions []string) (Fingerprints, error)
+}
+
+type Fingerprinter struct {
+	spinnerManager tui.ISpinnerManager
+}
+
+func NewFingerprinter() *Fingerprinter {
+	return &Fingerprinter{
+		spinnerManager: tui.NewSpinnerManager("Fingerprinting", "0"),
+	}
+}
+
+type FileFingerprint struct {
+	path          string
+	contentLength int64
+	fingerprint   []byte
+}
+
+func (f FileFingerprint) ToString() string {
+	// Replace backslashes with forward slashes to make the path platform independent
+	path := strings.ReplaceAll(f.path, "\\", "/")
+
+	return fmt.Sprintf("file=%x,%d,%s", f.fingerprint, f.contentLength, path)
+}
+func (f *Fingerprinter) FingerprintFiles(rootPath string, exclusions []string) (Fingerprints, error) {
+	log.Println("Warning: Fingerprinting is beta and may not work as expected.")
+	if len(rootPath) == 0 {
+		rootPath = filepath.Base("")
+	}
+
+	fingerprints := Fingerprints{}
+
+	f.spinnerManager.Start()
+	spinnerMessage := "files processed"
+	spinner := f.spinnerManager.AddSpinner(spinnerMessage)
+
+	nbFiles := 0
+
+	err := filepath.Walk(rootPath, func(path string, fileInfo os.FileInfo, err error) error {
+		nbFiles++
+
+		if err != nil {
+			return err
+		}
+
+		if !shouldProcessFile(fileInfo, exclusions, path) {
+			return nil
+		}
+
+		fingerprint, err := computeMD5(path)
+		if err != nil {
+			return err
+		}
+
+		fingerprints.Append(fingerprint)
+
+		if nbFiles%100 == 0 {
+			f.spinnerManager.SetSpinnerMessage(spinner, spinnerMessage, fmt.Sprintf("%d", nbFiles))
+		}
+
+		return nil
+	})
+
+	f.spinnerManager.SetSpinnerMessage(spinner, spinnerMessage, fmt.Sprintf("%d", nbFiles))
+
+	if err != nil {
+		spinner.Error()
+	} else {
+		spinner.Complete()
+	}
+
+	f.spinnerManager.Stop()
+
+	return fingerprints, err
+}
+
+func shouldProcessFile(fileInfo os.FileInfo, exclusions []string, path string) bool {
+	if fileInfo.IsDir() {
+		return false
+	}
+
+	if excluded(exclusions, path) {
+		return false
+	}
+
+	if isExcludedFile(path) {
+		return false
+	}
```

Comment:
Add check for symbolic/soft link -- currently fingerprint fails if any symlinks are present

Reply:
I have a fix for it now, could you test it out?

Reply:
Works for me now!